boms/cloud-lts-bom/pom.xml

```diff
@@ -256,6 +256,15 @@
         <artifactId>google-cloud-pubsub</artifactId>
         <version>${google.cloud.pubsub.version}</version>
       </dependency>
+
+      <!-- TODO: review this benefit and cost -->
+      <dependency>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>google-cloud-pubsub-bom</artifactId>
+        <version>${google.cloud.pubsub.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
```

Comment:
If we import pubsub-bom, java-pubsub test succeeds; otherwise it fails with dependency conflict. The lts-bom gets old `grpc-google-cloud-pubsub-v1:1.90.3`.

Reply:
But this has side-effect to Beam:

```
> Task :sdks:java:io:google-cloud-platform:test

org.apache.beam.sdk.io.gcp.pubsub.PubsubGrpcClientTest > publishOneMessage FAILED
    io.grpc.StatusRuntimeException at PubsubGrpcClientTest.java:200
        Caused by: java.lang.NoSuchFieldError at PubsubGrpcClientTest.java:200

org.apache.beam.sdk.io.gcp.pubsub.PubsubGrpcClientTest > pullOneMessage FAILED
    io.grpc.StatusRuntimeException at PubsubGrpcClientTest.java:145
        Caused by: java.lang.NoSuchFieldError at PubsubGrpcClientTest.java:145
```

Reply:
Beam uses `force` for each direct dependencies. The `force` function overrides the ones specified by the gcp-lts-bom.

Even if I apply the change:

```
suztomo-macbookpro44% git diff sdks/java/io/google-cloud-platform/build.gradle                                       
diff --git a/sdks/java/io/google-cloud-platform/build.gradle b/sdks/java/io/google-cloud-platform/build.gradle
index 7ac6878..57bd6d3 100644
--- a/sdks/java/io/google-cloud-platform/build.gradle
+++ b/sdks/java/io/google-cloud-platform/build.gradle
@@ -29,6 +29,8 @@ description = "Apache Beam :: SDKs :: Java :: IO :: Google Cloud Platform"
 ext.summary = "IO library to read and write Google Cloud Platform systems from Beam."
 
 dependencies {
+  compile enforcedPlatform("com.google.cloud:gcp-lts-bom:0.1.0-SNAPSHOT")
+  testRuntime enforcedPlatform("com.google.cloud:gcp-lts-bom:0.1.0-SNAPSHOT")
   compile enforcedPlatform(library.java.google_cloud_platform_libraries_bom)
   compile project(path: ":sdks:java:core", configuration: "shadow")
   compile project(":sdks:java:expansion-service")
suztomo-macbookpro44% ./gradlew  -PdisableSpotlessCheck=true :sdks:java:io:google-cloud-platform:dependencies |pbcopy
```

The testRuntimeClasspath uses Guava 25.1-jre

```
testRuntimeClasspath - Runtime classpath of source set 'test'.
+--- org.checkerframework:checker-qual:3.7.0
+--- com.google.cloud:gcp-lts-bom:0.1.0-SNAPSHOT
|    +--- com.google.guava:guava:30.1-jre -> 25.1-jre (c)
|    +--- com.google.protobuf:protobuf-java:3.15.6 -> 3.12.0 (c)
|    +--- io.grpc:grpc-alts:1.36.1 (c)
...
```


https://gist.github.com/suztomo/c965014ad2a06bf5780894d94c741ebb